proposals/nnnn-inferring-senable-for-methods.md

```diff
@@ -0,0 +1,221 @@
+
+
+```
+// Pseudo-code declaration of a Nominal Type:
+type NominalType {
+  func method(ArgType) -> ReturnType { /* body of method */ }
+}
+
+// Can desugar to these two global functions:
+func NominalType_method_partiallyAppliedTo(_ obj: NominalType) -> ((ArgType) -> ReturnType) {
+  let inner = { [obj] (_ arg1: ArgType) -> ReturnType in
+    return NominalType_method(obj, arg1)
+  }
+  return inner
+}
+// The actual method call
+func NominalType_method(_ self: NominalType, _ arg1: ArgType) -> ReturnType {
+  /* body of method */
+}
+```
+
+Thus, the only way a partially-applied method can be `@Sendable` is if the `inner` closure were `@Sendable`, which is true if and only if the nominal type conforms to `Sendable`.
+
+
+```
+type NominalType : Sendable {
+  func method(ArgType) -> ReturnType { /* body of method */ }
+}
+```
+
+For example, by declaring the following type `Sendable`, the partial and unapplied function values of the type would have implied Sendability and the following code would compile with no errors.
+```
+struct User : Sendable {
+  func updatePassword (new: String, old:String) -> Bool {
+    /* update password*/ 
+    return true
+  }
+}
+
+let unapplied: @Sendable (User) -> ((String, String) → Bool) = User.updatePassword // no error
+
+let partial: @Sendable (String, String) -> Bool = User().updatePassword // no error
+```
+
+
+
+## Detailed design
+
+This proposal includes four changes to `Sendable` behavior.
+
+The first two are what we just discussed regarding partial and unapplied methods.
+
+```
+struct User : Sendable {
+  var address
+  var password
+  
+  func changeAddress () {/*do work*/ }
+}
+```
+
+1. The inference of `@Sendable` for unapplied references to methods of a Sendable type. 
+
+```
+let unapplied : @Sendable (User) → ((String, String) → Void) = User.changeAddress // no error
+```
+
+2. The inference of `@Sendable` for partially-applied methods of a Sendable type.
+
+```
+let partial : @Sendable (String, String) → Void = User().changeAddress // no error
+```
+
+These two rules include partially applied and unapplied static methods but do not include partially applied or unapplied mutable methods. Unapplied references to mutable methods are not allowed in the language because they can lead to undefined behavior.  More details about this can be found in [SE-0042](https://github.com/apple/swift-evolution/blob/main/proposals/0042-flatten-method-types.md).
+
+Next is:
+
+3. The inference of `@Sendable`  when referencing non-local functions.
+
+Unlike closures, which retain the captured value, global functions can't capture any variables - because global variables are just referenced by the function without any ownership. With this in mind there is no reason not to make these `Sendable` by default. This change will also include static global functions.
```

Comment:
Nit: There's no such thing as a "static global function".
```suggestion
Unlike closures, which retain the captured value, global functions and static methods can't capture any variables - because global and static variables are just referenced by the function without any ownership. With this in mind there is no reason not to make these `Sendable` by default.
```

Reply:
What if I remove mention of static methods here? This section is just about global functions. The first two rules are about methods so I think it's better to talk about static methods there.